fog/test-client/src/test_client.rs

```diff
@@ -781,3 +864,125 @@ impl Drop for ReceiveTxWorker {
         }
     }
 }
+
+/// An object which tracks info about a Tx as it evolves, for logging context
+/// in case of errors.
+/// This is thread-safe so that we can share it with the receive worker
```

Comment:
Why not place the whole thing in a mutex?

Reply:
because it's more typing over all

Reply:
i guess it could just have one mutex inside of itself, that might be a little nicer